hierarchicalforecast/utils.py

```diff
@@ -70,22 +70,29 @@ def cov2corr(cov, return_std=False):
         return corr
 
 # %% ../nbs/utils.ipynb 9
-def _to_summing_matrix(S_df: pd.DataFrame):
+def _to_summing_matrix(S_df: pd.DataFrame, sparse_s: bool = False):
     """Transforms the DataFrame `df` of hierarchies to a summing matrix S."""
     categories = [S_df[col].unique() for col in S_df.columns]
     cat_sizes = [len(cats) for cats in categories]
     idx_bottom = np.argmax(cat_sizes)
     cats_bottom = categories[idx_bottom]
-    encoder = OneHotEncoder(categories=categories, sparse=False, dtype=np.float32)
+
+    try:
+        encoder = OneHotEncoder(categories=categories, sparse_output=sparse_s, dtype=np.float32)
+    except TypeError:  # sklearn < 1.2
+        encoder = OneHotEncoder(categories=categories, sparse=sparse_s, dtype=np.float32)
+
     S = encoder.fit_transform(S_df).T
+    # TODO: This likely still instantiates a dense matrix:
     S = pd.DataFrame(S, index=chain(*categories), columns=cats_bottom)
```

Comment:
I think we should use the [recommended constructor](https://pandas.pydata.org/docs/user_guide/sparse.html#interaction-with-scipy-sparse), something like:
```python
if sparse_s:
    df_constructor = pd.DataFrame.sparse.from_spmatrix
else:
    df_constructor = pd.DataFrame
S = df_constructor(S, index=chain(*categories), columns=cats_bottom)
```

Reply:
Done